interfaces/apparmor/backend.go

```diff
@@ -329,6 +330,19 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	if snapName == "core" && !release.OnClassic {
 		if li, err := filepath.Glob(filepath.Join(dirs.SystemApparmorCacheDir, "*")); err == nil {
 			for _, p := range li {
+				// On systems with unified cache directory,
+				// don't remove the snap profiles, only system
+				// and snap-confine profiles. snap-confine
+				// profiles are like the following:
+				// - usr.lib.snapd.snap-confine.real
+				// - usr.lib.snapd.snap-confine
+				// - snap.core.NNNN.usr.lib.snapd.snap-confine
+				// - snap-confine.core.NNNN
+				// TODO: also the "snapd" snap here soon
+				bn := path.Base(p)
+				if strings.HasPrefix(bn, "snap") && !strings.HasPrefix(bn, "snap-confine.core.") && !strings.Contains(bn, "usr.lib.snapd.snap-confine") {
```

Comment:
Could we please move this to a dedicated function? It would ease testing.

Reply:
Done